docs/database-functionalities/streams/implement-transformation-module.md

```diff
@@ -0,0 +1,298 @@
+---
+id: implement-transformation-module
+title: Implement transformation modules
+sidebar_label: Implement transformation modules
+---
+
+The prerequisite of connecting Memgraph to a Kafka stream is  to have a
+transformation module that can produce Cypher queries based on the received
+messages. We are going to implement a simple transformation that stores the
+properties of each message in a vertex.
+
+## Using Docker with transformation modules
+
+If you are using Docker to run Memgraph you will have to create a volume
+and mount it to access the `query_modules` directory. Yes, `query_modules`,
+because Memgraph can load transformations and query procedures from the same
+directory, even from the same module. Mounting a volume can be done by
+creating an empty directory `modules` and executing the following command:
+
+```shell
+docker volume create --driver local --opt type=none  --opt device=modules --opt o=bind modules
+```
+
+Now, you can start Memgraph and mount the created volume:
+
+```shell
+docker run -it --rm -v modules:/usr/lib/memgraph/query_modules -p 7687:7687 memgraph
+```
+
+Everything from the directory `/usr/lib/memgraph/query_modules` will be
+visible/editable in your mounted `modules` volume and vice versa.
+
+In the rest of this page `/usr/lib/memgraph/query_modules` will be used to
+refer to this directory. If you are using Docker, then please do the same but
+with the recently created `modules` directory.
+
+## Python API
+
+Transformations can be implemented using the Python API provided by Memgraph.
+If you wish to write your own transformation using the Python API, you need to
+have Python version 3.5.0 or above installed.
+
+Let's create a file called `transformation.py` in the
+`/usr/lib/memgraph/query_modules` directory. First,the foremost, import the
+`mgp` module, which contains definitions of the public Python API provided
+by Memgraph.
+
+```python
+import mgp
+```
+
+Next, we have to implement the function that does the transformation. For
+transformations the signature of the return type is fixed, while the signature
+of the transformation can vary. The whole signature of a transformation is the
+following:
```

Comment:
signature includes the return types so it's slightly awkward to read this.
```
For transformations, the return type is fixed, while the arguments can take one of the two supported forms.
The whole signature of a transformation is the following:
...
```

Reply:
```suggestion
transformations the return type is fixed, while the parameters
of the transformation can vary. The whole signature of a transformation is the
following:
```

Reply:
yep, that's fine.